src/Symfony/Component/AssetMapper/Compiler/JavaScriptImportPathCompiler.php

```diff
@@ -27,8 +27,18 @@
  */
 final class JavaScriptImportPathCompiler implements AssetCompilerInterface
 {
-    // https://regex101.com/r/qFoeoR/1
-    private const IMPORT_PATTERN = '/(?:\'(?:[^\'\\\\]|\\\\.)*\'|"(?:[^"\\\\]|\\\\.)*")|(?:import\s*(?:(?:\*\s*as\s+\w+|[\w\s{},*]+)\s*from\s*)?|\bimport\()\s*[\'"`](\.\/[^\'"`]+|(\.\.\/)*[^\'"`]+)[\'"`]\s*[;\)]?/m';
+    /**
+     * This regular expression matches:
+     * - Lines that start with comments //
+     * - Strings enclosed in single quotes
+     * - Strings enclosed in double quotes
+     * - Import statements
+     *
+     * Only the import statements are captured.
+     *
+     * @see https://regex101.com/r/1iBAIb/1
+     */
+    private const IMPORT_PATTERN = '/^(?:\/\/.*)|(?:\'(?:[^\'\\\\]|\\\\.)*\'|"(?:[^"\\\\]|\\\\.)*")|(?:import\s*(?:(?:\*\s*as\s+\w+|\s+[\w\s{},*]+)\s*from\s*)?|\bimport\()\s*[\'"`](\.\/[^\'"`]+|(\.\.\/)*[^\'"`]+)[\'"`]\s*[;\)]?/m';
```

Comment:
We can probably inline the comments directly in the regex (and add some spaces/new lines) to make the regex more easily understandable by using `/ms`.

Reply:
Is that better... or a bit too much ? 

```php
private const IMPORT_PATTERN = '/
    ^
        (?:\/\/.*)                     # Lines that start with comments
    |
        (?:
            \'(?:[^\'\\\\]|\\\\.)*\'   # Strings enclosed in single quotes
        |
            "(?:[^"\\\\]|\\\\.)*"      # Strings enclosed in double quotes
        )
    |
        (?:                            # Import statements (script captured)
            import\s*
                (?:
                    (?:\*\s*as\s+\w+|\s+[\w\s{},*]+)
                    \s*from\s*
                )?
        |
            \bimport\(
        )
        \s*[\'"`](\.\/[^\'"`]+|(\.\.\/)*[^\'"`]+)[\'"`]\s*[;\)]
    ?
/mx';
```

Reply:
Another solution can be : 

```php
private const PATTERN_LINES_START_WITH_COMMENTS = '(?:\/\/.*)';
private const PATTERN_STRINGS_ENCLOSED_SINGLE_OR_DOUBLE_QUOTES = '(?:
            \'(?:[^\'\\\\]|\\\\.)*\'   # Strings enclosed in single quotes
        |
            "(?:[^"\\\\]|\\\\.)*"      # Strings enclosed in double quotes
        )
';
```

And you concat everything in the ```IMPORT_PATTERN```. What do you think about it ?

Reply:
I genuinely have no opinion here, just happy if the next person to work with this pattern does not lose time to understand why and how we made this regexp :) So you guys tell me